selectrum.el

```diff
@@ -83,18 +96,12 @@ INPUT is a string, CANDIDATES is a list of strings."
   #'selectrum-default-candidate-refine-function
   "Function used to decide which candidates should be displayed.
 Receives two arguments, the user input (a string) and the list of
-candidates (strings) as returned by
-`selectrum-preprocess-candidates-function'. Returns a new list of
+candidates (strings) as given by
+`selectrum--preprocessed-candidates'. Returns a new list of
 candidates. Should not modify the input list. The returned list
 may be modified by Selectrum, so a copy of the input should be
 made. (Beware that `cl-remove-if' doesn't make a copy if there's
-nothing to remove.)
-
-Instead of a list of strings, may alternatively return an alist
-with the following keys:
-- `candidates': list of strings, as above
-- `input' (optional): transformed user input, used for
-  highlighting"
```

Comment:
Moving this means it's no longer possible to transform the user input unless you're dynamically generating candidates. Would it make more sense to keep this logic in the preprocessing step, applied after the optional dynamic candidate generation?

Reply:
Good idea, will do that later.

Reply:
Looking at this in more detail I'm not sure how to proceed. Do you have a use case in mind for changing the user input dynamically for a static collection?

The new update function could be used to update the input for static collections, too. It could  return the new input and the old candidates unchanged. In this case the update function should also take the candidates as an argument which it currently doesn't. Should I go this route or what are your thoughts on this?